core/conversion/converters/impl/pooling.cpp

```diff
@@ -355,6 +356,96 @@ auto pooling_registrations TRTORCH_UNUSED =
                  LOG_DEBUG("Output tensor shape: " << out_tensor->getDimensions());
                }
 
+               return true;
+             }})
+        .pattern(
+            {"aten::adaptive_max_pool2d(Tensor self, int[2] output_size) -> (Tensor, Tensor)",
+             [](ConversionCtx* ctx, const torch::jit::Node* n, args& args) -> bool {
+               auto in = args[0].ITensorOrFreeze(ctx);
+               auto in_shape = util::toVec(in->getDimensions());
+
+               if (in_shape.size() < 4) {
+                 auto new_shape = util::toDimsPad(in_shape, 4);
+                 LOG_DEBUG(
+                     "Input shape is less than 4D got: " << util::toDims(in_shape)
+                                                         << ", inserting shuffle layer to reshape to 4D tensor shape: "
+                                                         << new_shape);
+                 auto shuffle = ctx->net->addShuffle(*in);
+                 shuffle->setReshapeDimensions(new_shape);
+                 shuffle->setName((util::node_info(n) + " [Reshape to " + util::toStr(new_shape) + ']').c_str());
+                 in = shuffle->getOutput(0);
+                 in_shape = util::toVec(in->getDimensions());
+               }
+
+               auto out_size = util::toVec(util::toDims(args[1].unwrapToIntList()));
+
+               if (ctx->input_is_dynamic) {
+#if NV_TENSORRT_MAJOR < 7 || (NV_TENSORRT_MAJOR == 7 && NV_TENSORRT_MINOR < 1)
+                 LOG_WARNING(
+                     "Adaptive max pooling layer will be run through ATen, via not TensorRT, performace will be lower than expected. Consider switching either to static input shape or moving to non adaptive pooling if this is an issue");
+#else
+                 LOG_WARNING(
+                     "Adaptive max pooling layer will be run through ATen (on CPU), via not TensorRT, performace will suffer. Consider switching either to static input shape or moving to non adaptive pooling");
+#endif
+                 LOG_WARNING(
```

Comment:
Can you explain this warning?

Reply:
> Can you explain this warning?

I copy this from adaptive_avg_pool2d_plugin

Reply:
In core/conversion/converters/impl/plugins/interpolate_plugin.cpp: 191
If the tensorRT version is 7.0, adaptive pooling will run through cuda based aten, while tensorRT version > 7.0, the pooling will run throuhg cpu based aten